pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -220,6 +221,24 @@ def check_monitor_top_k(self, current):
 
         return monitor_op(current, self.best_k_models[self.kth_best_model_path])
 
+    @classmethod
+    def _format_checkpoint_name(cls, filename, epoch, metrics, prefix=""):
+        if not filename:
+            # filename is not set, use default name
+            filename = f'epoch={epoch}'
+        else:
+            # check and parse user passed keys in the string
+            groups = re.findall(r'(\{.*?)[:\}]', filename)
+            if len(groups):
+                metrics['epoch'] = epoch
+                for tmp in groups:
+                    name = tmp[1:]
+                    filename = filename.replace(tmp, name + '={' + name)
+                    if name not in metrics:
+                        metrics[name] = 0
+                filename = filename.format(**metrics)
```

Comment:
```suggestion
        if not filename:
            # filename is not set, use default name
            filename = '{epoch}'
        
        # check and parse user passed keys in the string
        groups = re.findall(r'(\{.*?)[:\}]', filename)
        if len(groups):
            metrics['epoch'] = epoch
            for tmp in groups:
                name = tmp[1:]
                filename = filename.replace(tmp, name + '={' + name)
                if name not in metrics:
                    metrics[name] = 0
            filename = filename.format(**metrics)
```
let regex do the formatting.